main/webapp/modules/core/scripts/project.js

```diff
@@ -591,6 +591,45 @@ Refine.getPermanentLink = function() {
   return "project?" + params.join("&");
 };
 
+(function() {
+  const systemInfo = {
+    availableMemory: 0,
+    totalMemory: 0,
+  };
+
+  systemInfo.byteToMB = function (bytes) {
+    return (bytes / 1024 / 1024).toFixed(2);
+  };
+
+  systemInfo.memoryWarn = function() {
+    return (this.availableMemory/this.totalMemory) * 100 < 5;
+  }
+
+  let intervalId;
+  intervalId = setInterval(function() {
+    $.ajax({
+      url: "command/core/get-memory-info?",
+      type: "GET",
+      async: false,
+      data: {
+      },
+      success: function(data) {
+        systemInfo.availableMemory = data["available_memory"];
+        systemInfo.totalMemory = data["total_memory"];
+      }
+    }).fail(function(jqXhr, textStatus, errorThrown ) {
+      alert( textStatus + ':' + errorThrown );
+      clearInterval(intervalId);
+    });
+
+    if(systemInfo.memoryWarn()) {
+      const msg = $.i18n('core-project/memory-warn');
+      var notification = new ProcessNotification(msg, 10000);
+      notification.display();
+    }
```

Comment:
This `if` block should rather be inside the `success` callback above, to avoid race conditions. With this code structure you would not even need to store the memory values in a global object (`systemInfo`).